bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java

```diff
@@ -654,6 +657,65 @@ public Collection<WorkItem> prRun(ScratchArea scratchArea) {
+                    // the actions done by the bot and the fixVersion should not be updated.
+                    var avoidForwardportMarker = "<!-- avoid forwardport " + issue.get().id() + " -->";
+                    var botUser = pr.repository().forge().currentUser();
+                    var avoidForwardportComment = comments.stream()
+                            .filter(c -> c.author().equals(botUser))
+                            .filter(c -> c.body().contains(avoidForwardportMarker))
+                            .findFirst();
+                    if (avoidForwardportComment.isEmpty()) {
+                        var parser = new OverridingJCheckConfigurationParser(pr.repository(),
+                                                                            bot.confOverrideRepository(),
+                                                                            bot.confOverrideName(),
+                                                                            bot.confOverrideRef());
+                        var jcheckConf = parser.parse(pr.targetHeadHash());
+                        var targetFixVersion = jcheckConf.flatMap(c -> c.general().version())
+                                                        .flatMap(v -> JdkVersion.parse(v));
+                        if (targetFixVersion.isPresent()) {
+                            var willCreateBackport = Backports.findIssue(issue.get(), targetFixVersion.get()).isEmpty();
+                            if (willCreateBackport) {
+                                var issueFixVersion = Backports.mainFixVersion(issue.get());
+                                if (issueFixVersion.isPresent() &&
+                                    targetFixVersion.get().compareTo(issueFixVersion.get()) > 0) {
+                                    log.info("Avoiding forward-backport of issue '" + issue.get().id() + "' with fixVersion '" + issueFixVersion.get().raw() +
+                                             "', repository fixVersion is '" + targetFixVersion.get().raw() + "'");
+                                    // This scenario would result in a "forward-backport" being created. Avoid it by
+                                    // updating the fixVersion of the original issue and creating a backport with
+                                    // the fixVersion from the original issue.
+                                    issue.get().setProperty("fixVersions",
+                                                            JSON.array().add(targetFixVersion.get().raw()));
+                                    var assignees = issue.get().assignees();
+                                    var backport = assignees.isEmpty() ?
+                                        Backports.createBackport(issue.get(), issueFixVersion.get().raw()) :
+                                        Backports.createBackport(issue.get(), issueFixVersion.get().raw(), assignees.get(0).username());
```

Comment:
The order of operations here risks leaving a mess if the bot is interrupted at some point. The fixVersion could be updated in the bug, but no backport created and no comment posted. On a rerun nothing would happen. The backport could be created but no comment be posted. On a rerun, no comment would be posted.

Operations like this need to be carefully crafted to guarantee that the end state is eventually reached. The simplest fix would be to create the backport first. Then we would also need to check for a backport with the same fixVersion as the main bug and only create the backport if that doesn't already exist. After that we change the fixVersion of the main bug.

That still won't guarantee that we will post the comment. For a complete solution we will need to post an interim comment with the wording "going to" added before performing any operations. We can look for this comment in addition to the other conditions to detect an incomplete existing attempt. Then when we have performed all operations, we finalize the comment by rewriting it.

We also need tests that verify that the end state is reached when the bot is presented with a PR in any of these incomplete states.

This may seem pedantic, but I'm very sure this will actually happen, more often than one might think. I went through a similar exercise with the /integrate command soon after taking over Skara because it was rather common for the bot to be interrupted mid integration.